src/urllib3/connection.py

```diff
@@ -420,19 +524,22 @@ def connect(self):
             # the TLS library, this cannot always be done. So we check whether
             # the TLS Library still thinks it's matching hostnames.
             cert = self.sock.getpeercert()
-            _match_hostname(cert, self.assert_hostname or server_hostname)
+            _match_hostname(cert, self.assert_hostname or server_hostname)  # type: ignore
```

Comment:
Why is `type: ignore` needed here?

Reply:
This is because `self.assert_hostname` could be `True`. In this case should `_match_hostname` be called with `server_hostname`? Or `assert_hostname` could be of type `Optional[Union[Literal['False'], str]]` if `typing_extensions` are added, so 'Literal` is available.

Reply:
Gotcha, related to https://github.com/urllib3/urllib3/issues/2126. And yeah let's use `Literal['False']` for now.

Reply:
Removed type ignore and changed `self.assert_hostname` type to include `Literal[False]` instead of `bool`